### DISCUSSION TITLE
Nonlinear Objective functions

### DISCUSSION THREAD

**Discussion by TELSER1** [scipopt/PySCIPOpt#849] [Q&A] [answered]
Hello,

I wanted to replicate the Mixed Integer Nonlinear Programming done in https://arxiv.org/pdf/2112.00278, which uses SCIP; I immediately noticed-modifying the objective in the default example to be 
`model.setObjective(x + y*y)`
that nonlinear objective functions are not supported.  

Before I go down the rabbit hole of trying to creation an extension, was there some fundamental difficulty in exposing this functionality, or was it simply a low priority?

Thanks for any help.

**Comment by Joao-Dionisio** [comment, marked as answer]
Hello @TELSER1! We had this discussion a couple of weeks ago. You can read about it on PR #781.

The bottom line is that you can use the ```set_nonlinear_objective``` function in the recipes package ([here](https://github.com/scipopt/PySCIPOpt/tree/master/src/pyscipopt/recipes)). It's quite simple, and it's not a part of SCIP proper, it just does an epigraph reformulation. It might not be the most efficient way of modeling the nonlinear objective, but it's pretty simple. 

Let me know if this helped!
